src/commands/clan/skills.ts

```diff
@@ -12,169 +12,185 @@ import {
   getRaidLog,
 } from "./_dread";
 
-const KILLMATCHER = /([A-Za-z0-9\-\_ ]+)\s+\(#\d+\)\s+defeated\D+(\d+)/;
-const SKILLMATCHER = /([A-Za-z0-9\-\_ ]+)\s+\(#\d+\)\s+used the machine/;
+const SKILL_KILL_MATCHER = /([A-Za-z0-9\-\_ ]+)\s+\(#\d+\)\s+(defeated\D+(\d+)|used the machine)/;
 
 export const data = new SlashCommandBuilder()
   .setName("skills")
   .setDescription("Get a list of everyone currently elgible for Dreadsylvania skills.");
 
-async function parseCurrentLogs(mapToUpdate: Map<string, PlayerData>) {
-  for (let clan of DREAD_CLANS) {
-    const raidLog = await getRaidLog(clan.id);
-    if (!raidLog) throw "Clan inaccessible";
-    addParticipationFromRaidLog(raidLog, mapToUpdate);
-  }
+export type Participation = { [username: string]: { skills: number; kills: number } };
+
+function mergeParticipation(a: Participation, b: Participation) {
+  return Object.entries(a).reduce(
+    (acc, [u, { skills, kills }]) => ({
+      ...acc,
+      [u]: {
+        skills: (acc[u]?.skills ?? 0) + skills,
+        kills: (acc[u]?.kills ?? 0) + kills,
+      },
+    }),
+    b
+  );
 }
 
-function addParticipantSkills(raidLog: string, mapToUpdate: Map<string, PlayerData>) {
-  const skillLines = raidLog.toLowerCase().match(new RegExp(SKILLMATCHER, "g"));
+async function getParticipationFromCurrentRaid() {
+  const raidLogs = await Promise.all(DREAD_CLANS.map((clan) => getRaidLog(clan.id)));
 
-  for (let skill of skillLines || []) {
-    const matchedSkill = skill.match(SKILLMATCHER);
-    if (!matchedSkill) continue;
+  return raidLogs.reduce(
+    (p, log) => mergeParticipation(p, getParticipationFromRaidLog(log)),
+    {} as Participation
+  );
+}
 
-    const participation = mapToUpdate.get(matchedSkill[1]);
+export function getParticipationFromRaidLog(raidLog: string) {
+  const lines = raidLog.toLowerCase().match(new RegExp(SKILL_KILL_MATCHER, "g")) || [];
+
+  return lines
+    .map((l) => l.match(SKILL_KILL_MATCHER))
+    .filter(notNull)
+    .map((m) => [m[1].toLowerCase(), m[3] ? "kills" : "skills", parseInt(m[3] || "1")] as const)
```

Comment:
you already `toLowerCase()`ed the entire raidlog, so you shouldn't have to apply it again to `m[1]`

In general I'm not personally jazzed about the paradigm of determining whether this is kills or skills based on whether or not we have a 3rd capture group

Reply:
I can text match the second capture group but it's much of a muchness. I'll change over to that

Reply:
Agreed about toLowerCase, this is because usernames always have to be lowercase and I was programming defensively